### PR TITLE
Feat/65/dashboard homepage design

### PR DESCRIPTION
**Pull request by VrushaliHaldankar01**
# PR Template Frontend

## One Line Description
Feat/65/dashboard homepage design
Designing of Home dashboard page of kids first

## UI/UX Outcome
- Add any relevant screenshots or videos to demonstrate the UI/UX changes.
- 
image

### CODE REVIEW

File: src/components/Dashboard/HomeDashboard/HomeDashboard.module.css [hunk 1]
```diff
@@ -0,0 +1,58 @@
+
+
+.dashboard{
```
Comment (AnnaSolovykh) [inline src/components/Dashboard/HomeDashboard/HomeDashboard.module.css:3]: To improve code readability and maintain consistency in our styling, we should add a space between the class selector and the opening brace.

File: src/components/Dashboard/HomeDashboard/HomeDashboard.module.css [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+  
+  
+  .content {
+    width: 100%;
+    background-color: #F2F5F5;
+  }
+ 
+    
+  .pagewindow{
```
Comment (AnnaSolovykh) [inline src/components/Dashboard/HomeDashboard/HomeDashboard.module.css:19]: The name` .pagewindow` does not adhere to our project's naming conventions. For consistency, it should be renamed to match our standard naming scheme. For example, if we use kebab-case for class names, it should be renamed to` .page-window.`

File: src/components/Dashboard/HomeDashboard/HomeDashboard.jsx [hunk 3]
```diff
@@ -1,30 +1,55 @@
...
+              <PendingEvents />
+            </Col>
+            {/* Message Notification */}
+            <Col>
+              <MessageNotification />
+            </Col>
+          </Row>
+        </Container>
+      ) : (
+        <div className={styles.pagewindow}>
```
Comment (AnnaSolovykh) [inline src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:36]: For containers we use Bootstrap containers - please refer to the projects' best practices.

File: src/components/shared/ui/Button/CustomButton.jsx [hunk 4]
```diff
@@ -41,3 +47,4 @@ CustomButton.propTypes = {
   link: PropTypes.string,
   children: PropTypes.any,
 };
+export default CustomButton;
```
Comment (AnnaSolovykh) [inline src/components/shared/ui/Button/CustomButton.jsx:50]: The code modification you added is outside the scope of your current ticket. Additionally, the CustomButton component is already exported, so the added export default CustomButton; is redundant and unnecessary.

File: src/components/shared/ui/Button/contentStyles.css [hunk 5]
```diff
@@ -40,3 +40,10 @@
   /* identical to box height, or 24px */
   text-align: center;
 }
+
+.icon-right-style {
```
Comment (AnnaSolovykh) [inline src/components/shared/ui/Button/contentStyles.css:44]: Before making changes to someone else's code, you should always seek permission and get their opinion. In this case, please ask @nandini92 to review this change.

File: src/pages/Dashboard/Dashboard.jsx [hunk 6]
```diff
@@ -21,6 +24,22 @@ const Dashboard = () => {
...
+            styles={`${styles['custom-button']} secondary-light`}
+            //styles='secondary-light'
+            iconRight={addIcon}
+            // iconRight='/path/to/icon/add.svg' // Ensure this path is correct
+            iconRightStyles='icon-right-style'
+            textStyles='text-style'
+          >
+            Add your kid
+          </CustomButton>
+        </div> */}
```
Comment (AnnaSolovykh) [inline src/pages/Dashboard/Dashboard.jsx:42]: Is there a reason to keep this commented-out code? If it's not needed, it should be removed to keep the codebase clean and maintainable. Additionally, please review the imports above and remove any that are unnecessary.

File: src/pages/Dashboard/Dashboard.module.css [hunk 7]
```diff
@@ -1,17 +1,46 @@
...
+
 }
 
 
 .content {
   width: 100%;
   background-color: #F2F5F5;
 }
+.pagewindow{
+
```
Comment (AnnaSolovykh) [inline src/pages/Dashboard/Dashboard.module.css:18]: The same here - this is css for the code which is not used and is commented out. Let's keep the code clean.

File: package.json [hunk 8]
```diff
@@ -22,6 +22,7 @@
     "@testing-library/user-event": "^14.5.2",
     "axios": "^1.7.2",
     "bootstrap": "^5.3.3",
+    "cors": "^2.8.5",
```
Comment (AnnaSolovykh) [inline package.json:25]: What was the purpose behind adding cors?

File: src/components/Dashboard/HomeDashboard/HomeDashboard.jsx [hunk 9]
```diff
@@ -1,30 +1,55 @@
...
+          <h1 className={styles.title}>
+            Create your kid’s profile and add all the necessary information by
+            clicking Add Your Kid button.
+          </h1>
+          <CustomButton
```
Comment (AnnaSolovykh) [inline src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:41]: When we click on the button we should be redirected to the Kid's page.

**Review by AnnaSolovykh — DISMISSED** [review]
Vrushali, great job! Please review the comments and make the changes needed. Also, please look into the icon issue. Its position and color don't match the Figma design. Please export the icon as SVG from Figma.
Screenshot 2024-08-01 at 09 23 25
Screenshot 2024-08-01 at 09 23 39

File: src/components/Dashboard/HomeDashboard/HomeDashboard.jsx [hunk 10]
```diff
@@ -1,30 +1,55 @@
...
 import WeeksOverview from './WeeksOverview';
 
 const HomeDashboard = () => {
-  return (
-    <Container fluid>
-      <Row>
-        {/* Weeks Overview */}
-        <WeeksOverview />
-      </Row>
+  const [showPart, setShowPart] = useState(false);
```
[the remark this answers is not in the document]
Comment (VrushaliHaldankar01) [reply src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:14]: Sure , ill edit it

File: src/components/Dashboard/HomeDashboard/HomeDashboard.jsx (hunk 3 above)
Comment (VrushaliHaldankar01) [reply src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:36]: Yes , this is the previous code , i did not modify it

Comment (VrushaliHaldankar01) [reply src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:36]: this is the static page code which was already there

Comment (AnnaSolovykh) [reply src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:36]: This is the code you have added. In the screenshot you can see code from the dev branch from the same component, and this code is using Bootstrap. Yours should also use it. 
Screenshot 2024-08-01 at 14 03 04

**Comment by VrushaliHaldankar01** [comment]
Made changes as per the request and also imported image as svg

**Comment by AnnaSolovykh** [comment]
[The solution of inheriting the color in svg on iterations](https://stackoverflow.com/questions/13000682/how-do-i-have-an-svg-image-inherit-colors-from-the-html-document)

**Review by pandjyoti — APPROVED** [review]

**Review by pandjyoti — APPROVED** [review]

File: src/components/Dashboard/HomeDashboard/HomeDashboard.module.css [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+  
+  .title {
+    width: 1200px;
+    height: 27px;
+    gap: 0px;
+    opacity: 0px;
+    font-family: var(--fontfamily);
+    font-size: var(--fontsizeS);
+    font-weight: 700;
+    line-height: 19.2px;
```
Comment (nandini92) [inline src/components/Dashboard/HomeDashboard/HomeDashboard.module.css:37]: Please remove these lines as they are redundant or have no effect:
    gap: 0px;
    opacity: 0px;
    font-family: var(--fontfamily);
    font-size: var(--fontsizeS);
        
     Note: I don't see any styling variables called fontfamily or fontsizeS.

File: src/components/shared/ui/Button/buttonStyles.css [hunk 12]
```diff
@@ -63,7 +63,8 @@
   justify-content: center;
   align-items: center;
   padding: 0px;
-
+  width: 18rem; /* 288px to rem */
+  height: 3.5rem; /* 56px to rem */
```
Comment (nandini92) [inline src/components/shared/ui/Button/buttonStyles.css]: Please remove these lines from buttonStyles and add to the custom-button styling in HomeDashboard.module.css

File: src/pages/Dashboard/Dashboard.jsx [hunk 13]
```diff
@@ -4,6 +4,9 @@ import { Outlet } from 'react-router-dom';
 import DashboardHeader from '@components/Dashboard/DashboardHeader/DashboardHeader';
 import Sidebar from '@components/Dashboard/Sidebar/Sidebar';
 
+// import CustomButton from '@components/shared/ui/Button/CustomButton';
+// import addIcon from '@components/shared/ui/Icon/add.svg';
+
```
Comment (nandini92) [inline src/pages/Dashboard/Dashboard.jsx]: Please remove commented out imports.

File: src/pages/ForgetPassword/ForgetPassword.jsx [hunk 14]
```diff
@@ -54,7 +54,7 @@ export default function ForgetPassword() {
     } catch (err) {
       //eslint-disable-next-line
       console.error(err);
-      setErrMsg('This email is not registered with us');
+      setErrMsg('There was an error with the request, please try again later.');
```
Comment (nandini92) [inline src/pages/ForgetPassword/ForgetPassword.jsx]: Figma only has this error:
image

Is this changed required?

File: src/components/Dashboard/HomeDashboard/HomeDashboard.jsx [hunk 15]
```diff
@@ -1,30 +1,61 @@
...
+            </Col>
+            {/* Message Notification */}
+            <Col>
+              <MessageNotification />
+            </Col>
+          </Row>
+        </Container>
+      ) : (
+        <Container fluid className={styles['page-window']}>
+          <h1 className={styles.title}>
```
Comment (nandini92) [inline src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:43]: I believe these should be <p></p> tags as h1 title is already set in header.

File: src/pages/Dashboard/Dashboard.module.css [hunk 16]
```diff
@@ -1,17 +1,32 @@
...
+  opacity: 0px;
+  font-family: var(--fontfamily);
+  font-size: var(--fontsizeS);
+  font-weight: 700;
+  line-height: 19.2px;
+  text-align: center;
+}
+.custom-button {
+  margin-top: 2rem; 
+}
```
Comment (nandini92) [inline src/pages/Dashboard/Dashboard.module.css]: Please remove this styling if it is not being used in the dashboard.

File: src/components/shared/ui/Button/contentStyles.css [hunk 17]
```diff
@@ -36,3 +37,4 @@
   /* identical to box height, or 24px */
   text-align: center;
 }
+
```
Comment (nandini92) [inline src/components/shared/ui/Button/contentStyles.css]: Please undo extra spacing in this file.

File: src/pages/Dashboard/Dashboard.module.css [hunk 18]
```diff
@@ -1,17 +1,32 @@
-.dashboard{
+
+
+.dashboard {
```
Comment (nandini92) [inline src/pages/Dashboard/Dashboard.module.css:1]: Please remove extra lines above .dashboard

**Review by nandini92 — CHANGES_REQUESTED** [review]

File: src/pages/Dashboard/Dashboard.jsx (hunk 13 above)
Comment (VrushaliHaldankar01) [reply src/pages/Dashboard/Dashboard.jsx]: Resolved

File: src/pages/ForgetPassword/ForgetPassword.jsx (hunk 14 above)
Comment (VrushaliHaldankar01) [reply src/pages/ForgetPassword/ForgetPassword.jsx]: This was taken by another person earlier so dont have clear idea of changes, cant modify it

File: src/pages/Dashboard/Dashboard.module.css (hunk 16 above)
Comment (VrushaliHaldankar01) [reply src/pages/Dashboard/Dashboard.module.css]: Resolved

File: src/components/shared/ui/Button/contentStyles.css (hunk 17 above)
Comment (VrushaliHaldankar01) [reply src/components/shared/ui/Button/contentStyles.css]: Resolved

File: src/pages/Dashboard/Dashboard.module.css (hunk 18 above)
Comment (VrushaliHaldankar01) [reply src/pages/Dashboard/Dashboard.module.css:1]: Resolved

File: src/components/shared/ui/Button/buttonStyles.css (hunk 12 above)
Comment (VrushaliHaldankar01) [reply src/components/shared/ui/Button/buttonStyles.css]: Resolved

File: src/components/Dashboard/HomeDashboard/HomeDashboard.jsx (hunk 15 above)
Comment (VrushaliHaldankar01) [reply src/components/Dashboard/HomeDashboard/HomeDashboard.jsx:43]: I guess , i imported the styles from css and in css i mentioned as title

File: src/components/Dashboard/HomeDashboard/HomeDashboard.module.css (hunk 11 above)
Comment (VrushaliHaldankar01) [reply src/components/Dashboard/HomeDashboard/HomeDashboard.module.css:37]: Resolved

File: package.json (hunk 8 above)
Comment (nandini92) [reply package.json]: @VrushaliHaldankar01  can you please review this? Do we need cors? If not please remove from package.json

File: src/components/Dashboard/HomeDashboard/HomeDashboard.module.css (hunk 11 above)
Comment (nandini92) [reply src/components/Dashboard/HomeDashboard/HomeDashboard.module.css:37]: @VrushaliHaldankar01  I still see the  gap and opacity setting.  We don't need gap and opacity setting for the title. 

Please also update the font-size to 1rem (16 px):
image

File: src/pages/ForgetPassword/ForgetPassword.jsx (hunk 14 above)
Comment (nandini92) [reply src/pages/ForgetPassword/ForgetPassword.jsx]: Can you change it back in your branch and push the changes?

Comment (VrushaliHaldankar01) [reply src/pages/ForgetPassword/ForgetPassword.jsx]: Yes  , ill do it

**Review by nandini92 — APPROVED** [review]

**Pull request merged**